src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -291,15 +291,66 @@ private bool StartCore(ProcessStartInfo startInfo)
                 }
             }
 
-            int childPid, stdinFd, stdoutFd, stderrFd;
+            int stdinFd = -1, stdoutFd = -1, stderrFd = -1;
             string[] envp = CreateEnvp(startInfo);
             string cwd = !string.IsNullOrWhiteSpace(startInfo.WorkingDirectory) ? startInfo.WorkingDirectory : null;
 
+            bool setCredentials = !string.IsNullOrEmpty(startInfo.UserName);
+            uint userId = 0;
+            uint groupId = 0;
+            if (setCredentials)
+            {
+                (userId, groupId) = GetUserAndGroupIds(startInfo);
+            }
+
             if (startInfo.UseShellExecute)
             {
+                string verb = startInfo.Verb;
+                if (string.IsNullOrEmpty(verb))
+                {
+                    // Default to 'open'.
+                    verb = "open";
+                }
+                else
+                {
+                    // Verbs are case-insensitive.
+                    verb = verb.ToLowerInvariant();
```

Comment:
Nitpicking again:
As I'm not a very experienced c# programmer, I didn't knew the difference between `ToLowerInvariant` and `ToLower`. When I looked it up, I found out, that one should use `ToUpperInvariant` instead (although I don't know if it makes a huge difference here).
[Source 1: Blog - How to correctly normalize strings and how to compare them (in .NET)](https://gingter.org/2018/07/10/how-to-correctly-normalize-strings-and-how-to-compare-them-in-net/)   
[Source 2: Microsoft - Best Practices for Using Strings in .NET](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#recommendations_for_string_usage)   
[Source 3: Microsoft - CA1308: Normalize strings to uppercase](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1308-normalize-strings-to-uppercase)